Alpaca-Finance-Agents/Large-Position-Alert/src/agent.ts

```diff
@@ -0,0 +1,124 @@
+export const workEventSig: string = VAULT_IFACE.getEvent('Work').format('sighash');
+
+function filterForVault(vaultAddresses: string[], address: string | null): [string, bigint | undefined] {
+  let filteredAddress: string = " ";
+  let filteredThreshold: bigint | undefined = BigInt(0);
+
+  // TODO: CONFIRM USING toLowerCase() IS NOT CAUSING SECONDARY NEGATIVE EFFECTS
+  if(address === vaultAddresses[0].toLowerCase()) {
+    filteredAddress = vaultAddresses[0];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[0])
+  } else if(address === vaultAddresses[1].toLowerCase()) {
+    filteredAddress = vaultAddresses[1];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[1])
+  } else if(address === vaultAddresses[2].toLowerCase()) {
+    filteredAddress = vaultAddresses[2];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[2])
+  } else if(address === vaultAddresses[3].toLowerCase()) {
+    filteredAddress = vaultAddresses[3];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[3])
+  } else if(address === vaultAddresses[4].toLowerCase()) {
+    filteredAddress = vaultAddresses[4];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[4])
+  } else if(address === vaultAddresses[5].toLowerCase()) {
+    filteredAddress = vaultAddresses[5];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[5])
+  }
+
+  return [filteredAddress, filteredThreshold];
+}
+
+const createFindingGenerator = (alertId: string): FindingGenerator => {
+  return (metadata: { [key: string]: any } | undefined): Finding => {
+    const decodedData = decodeParameters(
+      ["uint256", "uint256"],
+      metadata?.data
+    );
+
+    return Finding.fromObject({
+      name: "Large Position Event",
+      description: "Large Position Has Been Taken",
+      alertId: alertId,
+      severity: FindingSeverity.Info, // TODO: FIND OUT WHICH Severity TO USE HERE (Info?)
+      type: FindingType.Unknown,  // TODO: FIND OUT WHICH Type TO USE HERE (Unknown?)
+      metadata:{
+        positionId: decodedData[0],
+        borrowAmount: decodedData[1]
+      },
+    });
+  };
+};
+
+export function provideHandleTransaction(
+  alertId: string,
+  addresses: string[],
+): HandleTransaction {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    let [filteredAddress, filteredThreshold] = filterForVault(addresses, txEvent.to);
+
+    const handler = provideEventCheckerHandler(
+      createFindingGenerator(alertId),
+      workEventSig,
+      filteredAddress,
+      function filterLoanAmount(): boolean {
+
+        const filteredEvent = txEvent.filterEvent(workEventSig, filteredAddress);
+        const decodedData = decodeParameters(
+          ["uint256", "uint256"],
+          filteredEvent[0].data // TODO: CONFIRM THERE WILL ONLY BE ONE EVENT PER TXN, SINCE HARD CODING FIRST ITEM
+        );
+
+        if(filteredThreshold && decodedData[1] > filteredThreshold) {
+          return true;
+        } else {
+          return false;
+        }
+      }
+    );
```

Comment:
Here, you are missing multiple events, and you are also missing the event if the interaction was not directly with the vault emitting the event, because the only relevant address could be `txEvent.to`

Reply:
I would like to also suggest to avoid using `provideEventChecker` form `forta-agent-tools` because this template uses `txEvent.filterEvent` and that function is being deprecated. Instead of that try `txEvent.filterLog`